external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/DirectKafkaInputDStream.scala

```diff
@@ -67,6 +67,10 @@ private[spark] class DirectKafkaInputDStream[K, V](
     ekp
   }
 
+  val backpressureInitialRate: Long =
+    _ssc.sparkContext.conf.getLong("spark.streaming.backpressure.initialRate",
+      1 + _ssc.sparkContext.conf.getLong("spark.streaming.kafka.maxRatePerPartition", 1000L) / 10)
+
```

Comment:
It is to supply just a 10% of the maximum rate at the start of execution, while workers are still warming up. At least on my system this starting rate works nicely, gradually increasing the rate without building a backlog of stages.